apps/docs/app.lisp

```diff
@@ -22,46 +25,71 @@
 	handler)
 
 (defun populate-page (file)
-	(ui-root page_flow (Flow) (:flow_flags +flow_right_fill :font *env_window_font*
-			:color (get :color *window*))
-		(ui-label _ (:min_width +margin_width))
-		(ui-flow page (:flow_flags +flow_down_fill :min_width 800))
-		(ui-label _ (:min_width +margin_width)))
-	(defq state :text)
-	(each-line (lambda (line)
-			(task-slice)
-			(setq state ((handler-func state) state page (trim-end line (ascii-char 13)))))
-		(file-stream (cat "docs/" file ".md")))
-	((handler-func state) state page "")
-	(bind '(w h) (. page_flow :pref_size))
-	(. page_flow :change 0 0 w h)
-	(def page_scroll :min_width w)
-	(def (get :vslider page_scroll) :value (if (defq pos (. scroll_pos :find file)) pos 0))
-	(.-> page_scroll (:add_child page_flow) (:layout))
-	(.-> doc_flow :layout :dirty_all))
+	(when file
+		(ui-root page_flow (Flow) (:flow_flags +flow_right_fill :font *env_window_font*
+				:color (get :color *window*))
+			(ui-label _ (:min_width +margin_width))
+			(ui-flow page (:flow_flags +flow_down_fill :min_width 800))
+			(ui-label _ (:min_width +margin_width)))
+		(defq state :text)
+		(each-line (lambda (line)
+				(task-slice)
+				(setq state ((handler-func state) state page (trim-end line (ascii-char 13)))))
+			(file-stream file))
+		((handler-func state) state page "")
+		(bind '(w h) (. page_flow :pref_size))
+		(. page_flow :change 0 0 w h)
+		(def page_scroll :min_width w)
+		(def (get :vslider page_scroll) :value (if (defq pos (. scroll_pos :find file)) pos 0))
+		(.-> page_scroll (:add_child page_flow) (:layout))
+		(.-> doc_flow :layout :dirty_all)))
 
 (ui-window *window* (:color +argb_grey15)
-	(ui-title-bar _ "Docs" (0xea19) +event_close)
+	(ui-title-bar _ "Docs" (0xea19 0xea1b 0xea1a) +event_close)
 	(ui-flow doc_flow (:flow_flags +flow_right_fill :font *env_window_font* :color *env_toolbar_col*)
-		(ui-flow index (:flow_flags (logior +flow_flag_down +flow_flag_fillw))
-			(each (lambda (p)
-				(. (ui-button _
-					(:text p :flow_flags (logior +flow_flag_align_vcenter +flow_flag_align_hleft)))
-						:connect +event_button)) doc_list))
+		(ui-flow _ (:flow_flags +flow_stack_fill)
+			(ui-scroll *file_tree_scroll* +scroll_flag_vertical :nil
+				(. (ui-tree *file_tree* +event_file_folder_action
+						(:min_width 0 :color +argb_white :font *env_medium_terminal_font*))
+					:connect +event_tree_action))
+			(ui-backdrop _ (:color +argb_white)))
 		(ui-scroll page_scroll +scroll_flag_vertical (:min_height 900))))
 
+(defun select-node (file)
+	(when :t 
```

Comment:
should that be `(when file)` or maybe not there at all ?

Reply:
Done